ebos/eclproblem.hh

```diff
@@ -1021,8 +1007,10 @@ public:
         // write the desired VTK files.
         ParentType::writeOutput(verbose);
 
+        bool isSubStep = EWOMS_GET_PARAM(TypeTag, bool, EnableWriteAllSolutions) || !this->simulator().episodeWillBeOver();
```

Comment:
I don't think this does what you intended to achieve.

Suppose EnableWriteAllSolutions is true. Then isSubStep is always true. That, in turn, means that EclWriter::writeOutput() will treat every write request as if it is a substep (ministep) which ultimately means that no restart file will be created at all. You'll only get get summary output.

Reply:
I know. Just discovered this myself as well. Does the writer support BASIC=6? Then we could call the writer unconditionally and let the eclWriter handle this.

Reply:
Basic=6 has not been tested and (most probably) does not work.

Reply:
It seems like there is an issue with this. The restarts are written out every timesteps, but it seems like the last timesteps get replaced. Or just resinsight gets confused since the same episodeIndex is passed to the outputwriter.

Reply:
There seems to be some work left to support BASIC=6 in the eclWriter. I propose that we merge this fix for opm-simulators and fix opm-common later.